table-api/src/main/java/io/deephaven/api/agg/spec/AggSpec.java

```diff
@@ -20,150 +20,224 @@
         return AggSpecDistinct.of(includeNulls);
     }
 
     /**
+     * Returns the first value in the input column for each group.
+     *
      * @return {@link AggSpecFirst#of()}
      */
     static AggSpecFirst first() {
         return AggSpecFirst.of();
     }
 
     /**
+     * Specifies an aggregation that applies a formula to each input group to produce the corresponding output value.
+     *
+     * @param formula the formula to use to calculate output values from grouped input values
      * @return {@link AggSpecFormula#of(String)} for {@code formula}
      */
     static AggSpecFormula formula(String formula) {
         return AggSpecFormula.of(formula);
     }
 
     /**
+     * Specifies an aggregation that applies a formula to each input group to produce the corresponding output value.
+     * Each input column name is substituted for the param token for evaluation.
+     *
+     * @param formula the formula to use to calculate output values from grouped input values
      * @return {@link AggSpecFormula#of(String, String)} for {@code formula} and {@code paramToken}
      */
     static AggSpecFormula formula(String formula, String paramToken) {
         return AggSpecFormula.of(formula, paramToken);
     }
 
     /**
+     * Specifies an aggregation that freezes the first value for each group and ignores subsequent changes. When groups
+     * are removed, the corresponding output row is removedd. When groups are re-added (on a subsequent update cycle),
+     * the newly added value is then frozen.
+     *
      * @return {@link AggSpecFreeze#of()}
      */
     static AggSpecFreeze freeze() {
         return AggSpecFreeze.of();
     }
 
     /**
+     * Specifies an aggregation that outputs each group of input values as a Deephaven vector
+     * (io.deephaven.vector.Vector).
+     *
      * @return {@link AggSpecGroup#of()}
      */
     static AggSpecGroup group() {
         return AggSpecGroup.of();
     }
 
     /**
+     * Specifies an aggregation that outputs the last value in the input column for each group.
+     *
      * @return {@link AggSpecLast#of()}
      */
     static AggSpecLast last() {
         return AggSpecLast.of();
     }
 
     /**
+     * Specifies an aggregation that outputs the maximum value in the input column for each group. Only works for
+     * numeric or Comparable input types.
+     *
      * @return {@link AggSpecMax#of()}
      */
     static AggSpecMax max() {
         return AggSpecMax.of();
     }
 
     /**
+     * Specifier for a column aggregation that produces a median value from the input column's values for each group.
+     * Only works for numeric or Comparable input types.
+     *
      * @return {@link AggSpecMedian#of()}
      */
     static AggSpecMedian median() {
         return AggSpecMedian.of();
     }
 
     /**
+     * Specifier for a column aggregation that produces a median value from the input column's values for each group.
```

Comment:
I like this one as a template to apply to some of the ones that aren't quite up to this standard.